python/lsst/ts/standardscripts/maintel/m2/check_actuators.py

```diff
@@ -0,0 +1,267 @@
+        config : `types.SimpleNamespace`
+            Configuration
+        """
+
+        if self.mtcs is None:
+            self.mtcs = MTCS(
+                domain=self.domain,
+                log=self.log,
+            )
+            await self.mtcs.start_task
+
+        # Get list of M2 hardpoints from mtcs
+        # this should be async await?
+        self.hardpoint_ids = await self.mtcs.get_m2_hardpoints()
+
+        # Getting list of actuator ids
+        num_axial_actuator = NUM_ACTUATOR - NUM_TANGENT_LINK
+        self.axial_actuator_ids = list(
+            [
+                actuator_id
+                for actuator_id in np.arange(num_axial_actuator)
+                if actuator_id + 1 not in self.hardpoint_ids
+            ]
+        )
+
+        # Actuators that will be effectively tested, filtering out hardpoints
+        # (which are 1 indexed, csc is 0 indexed)
+        if config.actuators == "all":
+            self.actuators_to_test = self.axial_actuator_ids
+        else:
+            # check that no actuators are hardpoints
+            for actuator_id in config.actuators:
+                if actuator_id + 1 in self.hardpoint_ids:
+                    raise ValueError(
+                        f"Cannot run bump test on actuator {actuator_id},"
+                        "it is currently configured as a hardpoint"
+                    )
+                if actuator_id + 1 not in self.axial_actuator_ids:
+                    raise ValueError(
+                        f"Cannot run bump test on actuator {actuator_id}, it is not a valid axial actuator."
+                    )
+
+            self.actuators_to_test = config.actuators
+
+        if config.ignore_actuators:
+            self.actuators_to_test = [
+                actuator_id
+                for actuator_id in self.actuators_to_test
+                if actuator_id not in config.ignore_actuators
+            ]
+        if hasattr(config, "period"):
+            self.period = config.period
+        if hasattr(config, "force"):
+            self.force = config.force
+
+        await super().configure(config=config)
+
+    def set_metadata(self, metadata):
+        """Set metadata."""
+
+        metadata.duration = self.time_one_bump * (len(self.actuators_to_test))
+
+    async def run_block(self):
+        await self.assert_feasibility()
+        start_time = time.monotonic()
+
+        failed_actuators_id = []
+        for i, actuator_id in enumerate(self.actuators_to_test):
+            await self.mtcs.assert_all_enabled()
+
+            # Checkpoint
+            await self.checkpoint(
+                f"Running bump test on M2 Axial FA ID: {actuator_id} "
+            )
+            try:
+                await self.mtcs.run_m2_actuator_bump_test(
+                    actuator_id=actuator_id,
+                    period=self.period,
+                    force=self.force,
+                )
+            # raise a RuntimeError if the bump test fails
```

Comment:
I don't think this is right. The `self.mtcs.run_m2_actuator_bump_test` is calling the command directly. If the command fails it will raise a `salobj.base.AckError` or `base.AckTimeoutError` if the command timeout.